src/core/api/media_element_track_choice_manager.ts

```diff
@@ -53,21 +53,41 @@ interface IMediaElementTrackChoiceManagerEvents {
   textTrackChange: ITextTrack|null;
 }
 
+interface ICreatedAudioTrack {
+  track: IAudioTrack;
+  nativeTrack: ICompatAudioTrack;
+}
+
+interface ICreatedTextTrack {
+  track: ITextTrack;
+  nativeTrack: TextTrack;
+}
+
+interface ICreatedVideoTrack {
+  track: IVideoTrack;
+  nativeTrack: ICompatVideoTrack;
+}
+
+
+interface INativeTrackArray {
+  nativeTrack: ICompatVideoTrack|ICompatAudioTrack|TextTrack;
+}
+
 /**
  * Check if track array is different from an other one
  * @param {Array.<Object>} oldTrackArray
  * @param {Array.<Object>} newTrackArray
  * @returns {boolean}
  */
 function areTrackArraysDifferent(
-  oldTrackArray: Array<{ nativeTrack: ICompatVideoTrack|ICompatAudioTrack|TextTrack }>,
-  newTrackArray: Array<{ nativeTrack: ICompatVideoTrack|ICompatAudioTrack|TextTrack }>
+  oldTrackArray: INativeTrackArray[],
+  newTrackArray: INativeTrackArray[]
 ): boolean {
   if (newTrackArray.length !== oldTrackArray.length) {
     return true;
   }
   for (let i = 0; i < newTrackArray.length; i++) {
-    if (newTrackArray[i].nativeTrack !== oldTrackArray[i]?.nativeTrack) {
+    if (newTrackArray[i].nativeTrack !== oldTrackArray[i].nativeTrack) {
```

Comment:
I know that this guard against a case not possible NOW, but to me it's still safer to keep the question mark for very few disadvantages no?